File: CPP/HardDrive.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class HardDrive : public Komponen{
	private:
		int kapasitas_GB;
		string tipe_drive;
	
	public:
		HardDrive(){
			
		}
		
		HardDrive(int kapasitas_GB, string tipe_drive, string merk, string nama) : 
		Komponen(merk, nama)
		{
			this->kapasitas_GB = kapasitas_GB;
			this->tipe_drive = tipe_drive;
		}
		
		void setKapasitasGB(int kapasitas_GB){
			this->kapasitas_GB = kapasitas_GB;
		}
		
		void setTipeDrive(string tipe_drive){
			this->tipe_drive = tipe_drive;
		}
		
		int getKapasitasGB(){
			return this->kapasitas_GB;
		}
		
		string getTipeDrive(){
			return this->tipe_drive;
		}
		
		string display(){
			return getTipeDrive() + ' ' + getMerk() + ' ' + getNama() + " (" + to_string(getKapasitasGB()) + " GB) ";
		}
		
		~HardDrive(){
			
		}
};

File: Python/Program/HardDrive.py
from Komponen import Komponen

class HardDrive(Komponen):
    def __init__(self, kapasitas_GB=0, tipe_drive="", merk="", nama=""):
        super().__init__(merk, nama)
        self._kapasitas_GB = kapasitas_GB
        self._tipe_drive = tipe_drive

    def set_kapasitas_GB(self, kapasitas_GB):
        self._kapasitas_GB = kapasitas_GB

    def set_tipe_drive(self, tipe_drive):
        self._tipe_drive = tipe_drive

    def get_kapasitas_GB(self):
        return self._kapasitas_GB

    def get_tipe_drive(self):
        return self._tipe_drive

    def display(self):
        return f"{self.get_tipe_drive()} {self.get_merk()} {self.get_nama()} ({self.get_kapasitas_GB()} GB)"